src/rl/simple_proof_env.py

```diff
@@ -192,7 +192,10 @@ def reset(self):
         self._foward_to_lemma_proof()
         self.goal_start_time = time.time()
         self.inferences_used = 0
-        pass
+        
+        # If in Isabelle, automatically enter proof
+        if self.language == ProofAction.Language.ISABELLE:
+            self._dynamic_proof_executor.run_tactics(["proof -"])
```

Comment:
While Isabelle proofs can take many forms, we’ll limit the agent to those that start with proof - and end with qed. Otherwise, the dfs agent oftentimes ends up backtracking on proof -, which essentially makes the proof impossible to solve.

Here, we'll automatically apply the proof - tactic.